### PR TITLE
Regression

### PR DESCRIPTION
**Pull request by frankschae**
Implements Proposition 5.6 from issue #10 .

The sampling appears to be a slightly shifted. 

regression

### CODE REVIEW

**Comment by frankschae** [comment]
with different seed:
regression

**Comment by mschauer** [comment]
Wo wir aufpassen müssen: In einer früheren Version fehlt die Matrix `inv(a)` in der Gleichung für `mu`:
Screenshot 2020-11-16 at 11 06 19

Richtig ist 

Screenshot 2020-11-16 at 11 04 04

**Comment by frankschae** [comment]
with smaller noise value:

`par = [-0.2, 0.1, 0.09]`

regression

**Comment by frankschae** [comment]
tend = 20 Plot 
regression

File: test/regression_test.jl [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    for i in 1:length(Y)-1
+        ϕ = paramgrad(t, y)'
+        Gϕ = pinv(Y.prob.g(y, Y.prob.p, t)*Y.prob.g(y, Y.prob.p, t)')*ϕ # a is sigma*sigma'. Todo: smoothing like this is very slow
+        zi = ϕ'*Gϕ
+        t2, y2 = Y.t[i + 1], Y.u[i + 1]
+        dy = y2 - y
+        ds = t2 - t
+        #@show size(mu), size(Gϕ'), (dy - paramintercept(t, y)*ds)
+        mu = mu + Gϕ'*(dy - paramintercept(t, y)*ds)
```
Comment (frankschae) [inline test/regression_test.jl:75]: I think with `Gϕ'` here, we are implementing the correct version which contains `g(y, Y.prob.p, t)*g(y, Y.prob.p, t)'`

File: src/regression.jl [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    μ .= μ + Gϕ'*((y2 - y) - ϕ0*ds)
+    t = t2
+    y .= y2
+    Γ .= Γ + zi*ds
+  end
+  WW = Γ + Ξ
+  WL = (cholesky(Hermitian(WW)).U)'
+  th° = WL'\(randn(size(μ))+WL\μ)
+  return th°
```
Comment (mschauer) [inline src/regression.jl:32]: Lets return `Mitosis.Gaussian{(:F,:Γ)}(μ, WW)` here.

File: src/regression.jl [hunk 3]
```diff
@@ -0,0 +1,33 @@
+function conjugate(r::Regression, Y, Ξ)
+  @unpack k, fjac!, ϕ0func!, ϕ, μ, Γ, ϕ0, y, y2 = r
+  @unpack g, pest = k
+
+  t = Y.t[1]
+  y .= Y.u[1]
+
+  fjac!(ϕ, y, pest, t)
+  fill!(μ, zero(eltype(ϕ)))
```
Comment (mschauer) [inline src/regression.jl:9]: Let's allow `μ` and `Γ` arguments to `conjugate`, so several segments can be chained.

File: src/regression.jl (hunk 2 above)
Comment (mschauer) [reply src/regression.jl:32]: (I have to write the `rand` function for those still, though)

**Comment by mschauer** [comment]
Looks good.

regression

File: src/regression.jl (hunk 3 above)
Comment (frankschae) [reply src/regression.jl]: Added them as:

```.. , μ0=nothing, Γ0=nothing)```

so if not specified they will be filled with `zero` , if specified the values are written into `μ` and `Γ`.

File: src/regression.jl (hunk 2 above)
Comment (frankschae) [reply src/regression.jl]: I moved L30 and L31 to the tests and changed the return value.

**Pull request merged**